### ISSUE TITLE
y_true and true_boxes?

### ISSUE THREAD

**Issue by brainiac21** [experiencor/basic-yolo-keras#172]
Hi, I have some general questions about the YOLO here. what's the difference between y_true and true_boxes in the loss function? Does the y_true just contain more informations( like confidence ) than true_boxes?
Also is it possible to train the YOLO from the scratch without the pre-trained weights ? Thanks!

**Comment by experiencor** [comment]
@brainiac21 true_boxes and y_true contains exactly the same information (the bounding boxes) but in different forms. true_boxes can be inferred from y_true but the operations are complicated tensorflow functions. So I use both as a way to reduce the complexity of the computation of the custom loss.

I don't that it's possible at all as I have tried a few times. If you are lucky with your gradient descent, you may be able to.

**Comment by voqtuyen** [comment]
Could you explain in detail the forms of these twos?

**Comment by experiencor** [comment]
They contain exactly the same information but in different forms: the bounding boxes. This is to facilitate the computation of the loss function although it's a little confusing.

**Comment by voqtuyen** [comment]
Hi @experiencor 
y_true should be: (batch, 13, 13, 5, 5 + #classes) is that right? How about true_boxes?
true_boxes: shape=(1, 1, 1, TRUE_BOX_BUFFER, 4) ?

I still dont understand these 2 forms are different in dimensions, how can we use it to calculate:
https://github.com/experiencor/basic-yolo-keras/blob/435a6fb54796cc83691709f80a3298e5cfc55f92/frontend.py#L134

I'm still confused about this. Thanks

**Comment by experiencor** [comment]
TRUE_BOX_BUFFER = max number of boxes per images. What a bad name for this concept! I will change in next commits.

y_true has the shape of the output of the network. The information of the bounding boxes is embedded in this shape and play the role of the one-2-one mapping label of the network's output.

true_boxes is simply the list of all bounding boxes' dimensions: x, y, w, h.
